ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/state/PendingBlocks.java

```diff
@@ -27,24 +27,49 @@
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Collectors;
 
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.RemovalListener;
+
 public class PendingBlocks {
 
-  private final Map<Hash, Block> pendingBlocks = new ConcurrentHashMap<>();
+  // If more than 100 behind, Besu switch to full synchronization mode. 150 because it is possible
+  // to have multiple versions of the same block number
+  private static final int CACHE_PENDING_BLOCKS_SIZE = 150;
```

Comment:
Probably should make it clear in the variable name that this is a default like `DEFAULT_PENDING_BLOCKS_CACHE_SIZE`

Reply:
Removed